src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -165,6 +180,31 @@ public void batchExecute(List<IngestDocumentWrapper> ingestDocumentWrappers, Con
         });
     }
 
+    private Tuple<List<String>, Map<Integer, Integer>> sortByLengthAndReturnOriginalOrder(List<String> inferenceList) {
+        List<Tuple<Integer, String>> docsWithIndex = new ArrayList<>();
+        for (int i = 0; i < inferenceList.size(); ++i) {
+            docsWithIndex.add(Tuple.tuple(i, inferenceList.get(i)));
+        }
+        docsWithIndex.sort(Comparator.comparingInt(t -> t.v2().length()));
+        List<String> sortedInferenceList = docsWithIndex.stream().map(Tuple::v2).collect(Collectors.toList());
+        Map<Integer, Integer> originalOrderMap = new HashMap<>();
```

Comment:
suggestion: use an array or array list to store the map of order. i.e. element index -> value in array
It is more efficient than hashmap

Reply:
> It is more efficient than hashmap

I ran a test, read in List is more performant than Map, but for write, Map is better. The time difference for 10000 times run with data size 1000, for write, Map takes 0.08s less, for read, List takes 0.05s less. For this logic read and write is 1:1, it's really hard to tell which one is better than the other.

Reply:
I would double check on performance, basically based on:
- list still is for wrapper type Integer, that boxing/unboxing takes cycles. I think we can use array of ints here
- 1.000 is not that big size, datasets can be in tens of hundreds of millions